File: frontend/src/app/model/flight/SearchFlightsParams.ts
import {NumberOfStops} from "./NumberOfStops";

export class SearchFlightsParams {
  private seatClass: string;
  private from: string;
  private to: string;
  private numberOfChildren: number;
  private numberOfAdults: number;
  private selectedNumberOfStops: string;
  private returnTicket: boolean;
  private singleDate: Date;
  private dateRangeStart: Date;
  private dateRangeEnd: Date;

  constructor(
    seatClass: string,
    from: string,
    to: string,
    numberOfChildren: number,
    numberOfAdults: number,
    selectedNumberOfStops: string,
    returnTicket: boolean,
    singleDate: Date,
    dateRangeStart: Date,
    dateRangeEnd: Date
  ) {
    this.seatClass = seatClass;
    this.from = from;
    this.to = to;
    this.numberOfChildren = numberOfChildren;
    this.numberOfAdults = numberOfAdults;
    this.selectedNumberOfStops = selectedNumberOfStops;
    this.returnTicket = returnTicket;
    this.singleDate = singleDate;
    this.dateRangeStart = dateRangeStart;
    this.dateRangeEnd = dateRangeEnd;
  }
}

File: backend/service/src/main/java/com/ftn/sbnz/backward/service/dto/SearchFlightsParams.java
package com.ftn.sbnz.backward.service.dto;

import com.ftn.sbnz.backward.model.models.flight.NumberOfStops;
import com.ftn.sbnz.backward.model.models.flight.SeatClass;
import lombok.*;

import java.time.Instant;
import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Data
public class SearchFlightsParams {
    private SeatClass seatClass;
    private String from;
    private String to;
    private int numberOfChildren;
    private int numberOfAdults;
    private NumberOfStops selectedNumberOfStops;
    private boolean returnTicket;
    private Instant singleDate;
    private Instant dateRangeStart;
    private Instant dateRangeEnd;
}
